src/Clock.jsx

```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+export class Clock extends React.Component {
+  static propTypes = {
+    time: PropTypes.string.isRequired,
+    visibility: PropTypes.bool.isRequired,
+    name: PropTypes.number.isRequired,
+  };
+
+  componentDidUpdate(prevProps) {
+    if (prevProps.name !== this.props.name) {
+      // eslint-disable-next-line
+      console.log(`The Clock was renamed from ${
+        prevProps.name} to ${
+        this.props.name}`);
+    }
```

Comment:
You could just only console.log your new name directly when you clicking the button (right after making a new random name). No need in componentDidUpdate